### PR TITLE
[LICM] Fix dropped metadata

### LINKED ISSUE #91919: LICM drop call instruction metadata
In LICM.cpp::cloneInstructionInExitBlock, we have a special treatment for call instructions. However, in the end, when we do `New = CallInst::Create(CI, OpBundles);`, therefore original call instruction metadata isn't cloned. We probably need to add `New->copyMetadata(*CI);` after the call instruction creation.

Reproducer:
https://godbolt.org/z/MqTY63KW7

You can see that the 'alias.scope' and 'noalias' metadata were lost.

### PR DESCRIPTION
**Pull request by tgymnich**
Fixes  #91919

LICM drops metadata for call instructions when cloning instructions.
This patch just adds the missing `copyMetadata`.

### CODE REVIEW

**Review by nikic — CHANGES_REQUESTED** [review]
Please add a test case in llvm/test/Transforms/LICM.

**Review by nikic — APPROVED** [review]
LGTM

File: llvm/test/Transforms/LICM/licm-ci.ll [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  %i = phi i32 [ 0, %entry ], [ %i.next, %loop ]
+  %gep = getelementptr <4 x i16>, ptr %in, i32 %i
+  %load = call <4 x i16> @llvm.masked.load.v4i16.p0(ptr %gep, i32 2, <4 x i1> <i1 true, i1 false, i1 true, i1 false>, <4 x i16> <i16 0, i16 poison, i16 0, i16 poison>), !alias.scope !2, !noalias !2
+  %reduce = call i16 @llvm.vector.reduce.add.v4i16(<4 x i16> %load)
+  %i.next = add i32 %i, 1
+  %cmp = icmp ult i32 %i.next, 10
+  br i1 %cmp, label %loop, label %exit
+
+exit:                               ; preds = %loop
+  store i16 %reduce, ptr %out
```
Comment (nikic) [inline llvm/test/Transforms/LICM/licm-ci.ll:34]: Can simplify this a bit more by replacing this store+param with a return.

**Comment by tgymnich** [comment]
deleted the branch too early. Only saw the approval

File: llvm/test/Transforms/LICM/licm-ci.ll (hunk 1 above)
Comment (tgymnich) [reply llvm/test/Transforms/LICM/licm-ci.ll]: done. Sorry for the delay

**Pull request merged**